test/powershell/Modules/Microsoft.PowerShell.Utility/Remove-Alias.Tests.ps1

```diff
@@ -0,0 +1,58 @@
+Describe "Remove-Alias" -Tags "CI" {
+    It "Remove-Alias should remove a non-readonly alias"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -ErrorAction Stop
+            Remove-Alias -Name "tral" -ErrorAction Stop
+            Get-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should throw on a readonly alias"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -Option ReadOnly -ErrorAction Stop
+            Remove-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'AliasNotRemovable,Microsoft.PowerShell.Commands.RemoveAliasCommand'
+    }
+
+    It "Remove-Alias should remove a non-readonly alias with force"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -ErrorAction Stop
+            Remove-Alias -Name "tral" -Force -ErrorAction Stop
+            Get-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should remove a readonly alias with force"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -Option ReadOnly -ErrorAction Stop
+            Remove-Alias -Name "tral" -Force -ErrorAction Stop
+            Get-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+    }
+
+    It "Remove-Alias should throw if alias does not exist"{
+        {
+            Get-Alias -Name "tral" -ErrorAction SilentlyContinue | Should BeNullorEmpty
+            Remove-Alias -Name "tral" -ErrorAction Stop
+        } | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.RemoveAliasCommand'
+    }
+
+    It "Remove-Alias should remove multiple alias at once"{
+        {
+            Set-Alias -Name "tral" -Value "Remove-Alias" -ErrorAction Stop
+            Set-Alias -Name "tral2" -Value "Remove-Alias" -ErrorAction Stop
+            Set-Alias -Name "tral3" -Value "Remove-Alias" -ErrorAction Stop
+            Remove-Alias -Name "tral","tral2","tral3" -ErrorAction Stop
+            Get-Alias -Name "tral" -ErrorAction Stop | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+            Get-Alias -Name "tral2" -ErrorAction Stop | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+            Get-Alias -Name "tral3" -ErrorAction Stop | ShouldBeErrorId 'ItemNotFoundException,Microsoft.PowerShell.Commands.GetAliasCommand'
+        }
+    }
+
+    It "Remove-Alias should throw on out-of-range scope"{
```

Comment:
it might be worth adding some tests with explicit scope probes. Say create an alias in scope 3, then remove it without a scope, it should still be in scope 3, etc. I don't think we have many tests which explicitly loot at testing our scopes, so it would be a welcome change

Reply:
I see that CodeCov reports `internal void RemoveAlias` falls under tests. Formally we don't get increasing code coverage with the change.
Although I found up to 34 test files with "scope" word in the repo the SessionStateAliasAPIs.cs code coverage is only 68%. I've view these tests before, some look like tricky so I didn't venture to make improvements. I think we need to open new tracking Issue if we want have better scope tests.